packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -18,11 +18,11 @@ const selectors = {
 	paragraphBlocks: 'p.block-editor-rich-text__editable',
 
 	// Top bar selectors.
-	editPostHeader: '.edit-post-header',
-	publishPanelToggle: '.editor-post-publish-panel__toggle',
+	postToolbar: '.edit-post-header',
 	settingsToggle: '[aria-label="Settings"]',
+	saveDraftButton: '.editor-post-save-draft',
```

Comment:
This element unfortunately changes attributes depending on the viewport :sigh:

In the mobile viewport, this element has an `[aria-label="Save draft"]`. In desktop viewport, it is no longer present.

In desktop viewport, the text `Save draft` is present, but in the mobile viewport, the text  is not present.

Rather than fight it, using the CSS class is the easiest solution.

Reply:
I think we should still be able to select by text? e.g. `button:text("Save draft")`

Reply:
I think we should still be able to select by text? e.g. `button:text("Save draft")`

Reply:
Unfortunately that is not the case as seen in the following screencap:

![image](https://user-images.githubusercontent.com/6549265/129416485-d8893b75-70f3-4c4c-9b8a-b3e5ec2cc899.png)

The first call was made with the desktop viewport, and the second call was made with mobile viewport.
The `:text(...)` pseudo-class only works with visible text. For some inexplicable reason, the `aria-label` attribute is not consistently applied to this element, so the only common selector that works across desktop and mobile for this would be the `.editor-post-save-draft` CSS class.